code/modules/client/preferences.dm

```diff
@@ -343,6 +345,40 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		value_cache -= preference.type
 		preference.apply_to_client(parent, read_preference(preference.type))
 
+
+/datum/preferences/proc/get_ui_asset_typepaths()
+	var/list/asset_typepaths = list(
+		/datum/asset/spritesheet/preferences,
+		/datum/asset/json/preferences,
+	)
+
+	for (var/datum/preference_middleware/preference_middleware as anything in middleware)
+		asset_typepaths += preference_middleware.get_ui_asset_typepaths()
+
+	return asset_typepaths
+
+/// Check if UI assets are ready
+/datum/preferences/proc/are_assets_ready()
+	return asset_loading_count == 0
+
+/datum/preferences/proc/register_assets_loaded()
+	for(var/typepath in get_ui_asset_typepaths())
+		var/datum/asset/asset = get_unregistered_asset_datum(typepath)
+		if(!asset.is_ready())
+			++asset_loading_count
+			RegisterSignal(asset, COMSIG_ASSET_GENERATED, PROC_REF(asset_loaded))
+
+		// we want to ensure that all our spritesheets have prioritized loading
```

Comment:
Why?

Reply:
Well rather, spritesheets sure, but not all files. We send JSONs that don't need to be early assets because the point of early assets is to run before atoms in CI.